backend/api/CourseApi/index.js

```diff
@@ -10,15 +10,45 @@ import CourseModel from '~/models/CourseModel';
 router.get('/public', catchAsync(async (request, response) => {
   const pageSize = request.query.pageSize;
   const pageNumber = request.query.pageNumber;
+
+
+  // Conversion effort -  Raw SQL to Knex like syntax 
+  // Can be used later when support for some functions like count, camelize column names are available from knex
+
+  // let query = knex().select(knex.raw('row_to_json(course.*) AS course, row_to_json("user".*) AS author, row_to_json(course_category.*) AS course_category, COUNT(distinct course_user_is_learning.user_id) AS amount_of_users_learning_this_course, ROUND(AVG(course_rating.rating),1) AS average_course_rating, COUNT(distinct problem.id) AS amount_of_problems, COUNT(distinct course_rating.id) AS amount_of_course_ratings'))
+  // .from('course')
+  // .innerJoin('problem', 'problem.course_id', '=', 'course.id')
+  // .innerJoin('user', 'course.user_id', '=', 'user.id')
+  // .innerJoin('course_category', 'course.course_category_id', '=', 'course_category.id')
+  // .leftOuterJoin('course_rating', 'course_rating.course_id', 'course.id')
+  // .leftOuterJoin(knex.raw('course_user_is_learning ON (course_user_is_learning.active = true AND course.id = course_user_is_learning.course_id)'))
+  // .where('course.if_public', 'true')
+  // .andWhere(2, "<=", subquery)
+  // .groupByRaw('course.id, "user".id, course_category.id')
+  // .orderByRaw(sortByQuery) // This function to be copied from model
+  // pageSize ? query.limit(pageSize) : '';
+  // pageNumber && pageSize ? query.offset((pageNumber - 1) * pageSize) : '';
+  
+  // const OnePageOfCourses = await query;
+
   const onePageOfCourses = await CourseModel.select.allPublic({
     sortBy: request.query.sortBy,
     limit: pageSize,
     offset: (pageNumber - 1) * pageSize,
     courseCategoryId: request.query.courseCategoryId
   });
-  const amountOfAllCourses = await CourseModel.select.countAllPublic({
-    courseCategoryId: request.query.courseCategoryId
-  });
+
+  const courseCategoryId = request.query.courseCategoryId;
+  const subquery = knex('problem').count('id').whereRaw('problem.course_id = course.id');
+  let  amountofAllCoursesQuery = knex('course').count('id', {as: 'amount_of_public_courses'})
+  .where('if_public', 'true')
+  .andWhere(2, "<=", subquery);
+
+  if(courseCategoryId) {
+    amountofAllCoursesQuery = amountofAllCoursesQuery.andWhere('course_category_id', courseCategoryId);
+  }
+
+  const amountOfAllCourses = await amountofAllCoursesQuery;
```

Comment:
Does this actually work locally? Do we not need to access `'amount_of_public_courses'` somehow?

Reply:
Sorry, my bad. I did not push the latest change. Done now. Please check.